.travis.yml

```diff
@@ -0,0 +1,104 @@
+
+language: generic
+sudo: required
+dist: trusty
+
+matrix:
+  fast_finish: true
+
+  include:
+    - os: linux
+      addons:
+        apt:
+          sources: ['ubuntu-toolchain-r-test']
+          packages: ['g++-6', 'cmake', 'python3', 'python3-pip', 
+'python3-setuptools', 'nodejs', 'npm', 'libsfml-dev', 'libprotobuf-dev', 
+'protobuf-compiler', 'libzmq-dev']
+      env: CCOMPILER='gcc-6' CXXCOMPILER='g++-6'
+
+    - os: linux
+      addons:
+        apt:
+          sources: ['ubuntu-toolchain-r-test']
+          packages: ['g++-5', 'cmake', 'python3', 'python3-pip', 
+'python3-setuptools', 'nodejs', 'npm', 'libsfml-dev', 'libprotobuf-dev', 
+'protobuf-compiler', 'libzmq-dev']
+      env: CCOMPILER='gcc-5' CXXCOMPILER='g++-5'
+
+    - os: linux
+      addons:
+        apt:
+          sources: ['llvm-toolchain-trusty-3.9']
+          packages: ['clang-3.9', 'cmake', 'python3', 'python3-pip', 
+'python3-setuptools', 'nodejs', 'npm', 'libsfml-dev', 'libprotobuf-dev', 
+'protobuf-compiler', 'libzmq-dev']
+      env: CCOMPILER='clang-3.9' CXXCOMPILER='clang++-3.9'
+
+    - os: linux
+      addons:
+        apt:
+          sources: ['llvm-toolchain-trusty-4.0']
+          packages: ['clang-4.0', 'cmake', 'python3', 'python3-pip', 
+'python3-setuptools', 'nodejs', 'npm', 'libsfml-dev', 'libprotobuf-dev', 
+'protobuf-compiler', 'libzmq-dev']
+      env: CCOMPILER='clang-4.0' CXXCOMPILER='clang++-4.0'
+
+    - os: osx
+      osx_image: xcode8.2
+      node_js: 4
+      env: CCOMPILER='clang' CXXCOMPILER='clang++'
+
+    - os: osx
+      osx_image: xcode8.2
+      node_js: 6
+      env: CCOMPILER='clang' CXXCOMPILER='clang++'
+
+before_install:
+  - export CC=${CCOMPILER} CXX=${CXXCOMPILER}
+  - if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then sudo 
+update-alternatives --install /usr/bin/node node /usr/bin/nodejs 10; fi
+  - if [[ "${TRAVIS_OS_NAME}" == "osx" ]]; then brew update && brew 
+install python3; fi
+
+install:
+  # Make sure building libsweep works
```

Comment:
You should probably remove everything from here on and replace it with compiling your cpp files only.

The idea is that Travis will show you if your snippets compile on gcc / clang for every commit you make.